Integrate Azure Functions deployment with In-Vitro (#569)
vhive-serverless/invitro: docs/loader.md

  - Under `Manage Quota`, select `AWS Lambda` service and click `View quotas` (Alternatively, click [here](https://us-east-1.console.aws.amazon.com/servicequotas/home/services/lambda/quotas))
  - Under `Quota name`, select `Concurrent executions` and click `Request increase at account level` (Alternatively, click [here](https://us-east-1.console.aws.amazon.com/servicequotas/home/services/lambda/quotas/L-B99A9384))
  - Under `Increase quota value`, input `1000` and click `Request`
  - Await AWS Support Team to approve the request. The request may take several days or weeks to be approved.

## Using Azure Functions

**Pre-requisites:**
1. Microsoft Azure account with an active subscription ID
2. Go installed
3. Python3 installed

**Quick Setup for Azure Deployment:**
1. Install the Azure CLI and verify installation:
    ```bash
    curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
    az --version
    ```
2. Log in as a user (Note: This will open a browser window to select Azure account):
    ```bash
    az login

>     ```

Reviewer: This is not always possible. There are no browser available on cloudlab nodes. Can you provide pure CLI method of logging in?

3. Create an Azure Service Principal: 
   ```bash
   az ad sp create-for-rbac --name "InVitro" --role Contributor --scopes /subscriptions/<your-subscription-id>
   ```
4. Set the following environment values: 
   ```bash
   export AZURE_APP_ID=<appId>
   export AZURE_PASSWORD=<password>
   export AZURE_TENANT=<tenant>
   ```
5. Use the Service Principal credentials in order to run automated scripts without manual login: 
   ```bash
   az login --service-principal --username $AZURE_APP_ID --password $AZURE_PASSWORD --tenant $AZURE_TENANT
   ```
6. Start the Azure Functions deployment experiment:
    ```bash
    go run cmd/loader.go --config cmd/config_azure_trace.json
    ```
---
Note:

> - Current deployment is via ZIP

Reviewer: Please add the note that Go is unsupported in Consumption Plan (or whatever we use) and we are using Python instead

  - Under `Increase quota value`, input `1000` and click `Request`
  - Await AWS Support Team to approve the request. The request may take several days or weeks to be approved.

## Using Azure Functions

**Pre-requisites:**
1. Microsoft Azure account with an active subscription ID
2. Existing Service Principal for authentication (refer to Notes section)
3. Go installed
4. Python3 installed

**Quick Setup for Azure Deployment:**
1. Install the Azure CLI and verify installation:
    ```bash
    curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
    az --version
    ```
2. Use existing Service Principal credentials in order login to Azure.
    ```bash
    az login --service-principal --username $AZURE_APP_ID --password $AZURE_PASSWORD --tenant $AZURE_TENANT
    ```
   > Refer to Note section for generation of Service Principal credentials
3. Start the Azure Functions deployment experiment:
    ```bash
    go run cmd/loader.go --config cmd/config_azure_trace.json
    ```
---
Notes:

> - Service Principal must be created before running experiment, as some environments do not have browsers (e.g. CloudLab). Perform these steps in an environment that allows launching of browser and use the generated credentials. 

Reviewer: ```suggestion
- Service Principal must be created before running the experiment, as some environments do not GUI (e.g., CloudLab nodes). You can perform these steps in an environment that allows the launching of the browser and use the generated credentials. 
```

Overall, can this be done via Azure web console? This is much simpler than doing it via CLI, since this part you don't need to do in experiment environment.
